### PR TITLE
add a Gaussian temperature interpolation for the plume feature

### PR DESCRIPTION
**Pull request by jdannberg**
I am here adding two images of what the test cases would look like with a grid file (but I thought they'd be faster to run with gwd-dat):

Cartesian:
plume_gaussian

Spherical:
plume_spherical

### CODE REVIEW

**Comment by MFraters** [comment]
The pictures look really cool! I will do my best to do a review in the coming week.

File: source/world_builder/features/plume_models/temperature/gaussian.cc [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                            "the plume as given by the plume feature) can be changed with depth. "
+                            "Temperature is always interpolated in a horizonzal/radial plane, except for the plume "
+                            "head: If the first depth of the plume centerline and the minimum depth of the plume "
+                            "feature are different, an ellipsoidal plume head is created in this depth range. "
+                            "Within this plume head, temperature is interpolated radially, i.e., depending on the "
+                            "distance from the center of the ellipsoid.");
+
+          prm.declare_entry("depths", Types::Array(Types::Double(0)),
+                            "The temperature at the center of this feature in degree Kelvin."
+                            "If the value is below zero, the an adiabatic temperature is used.");
```
Comment (MFraters) [inline source/world_builder/features/plume_models/temperature/gaussian.cc:72]: I don't think this description is correct.

File: source/world_builder/features/plume_models/temperature/gaussian.cc [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                            "The sigma (standard deviation) of the Gaussian function used to compute the "
+                            "temperature distribution within the plume. This sigma is non-dimensinal, i.e. "
+                            "it is defined relative to the distance between the plume center and margin as "
+                            "defined by the plume feature. Choosing a sigma of 1 therefore means that the "
+                            "temperature at the plume margin is set to a fraction of 1/sqrt(e) (approx. 0.61) "
+                            "of the centerline temperature. To achieve a smoother transition between the "
+                            "plume temperature and the outside temperature a smaller values has to be chosen "
+                            "for the gaussian sigmas.");
+
+          // TODO: assert that the three have the same length
```
Comment (MFraters) [inline source/world_builder/features/plume_models/temperature/gaussian.cc:86]: This todo belongs in parse_entries, and it looks like it is already done. I think you can just remove this.

**Review by MFraters — APPROVED** [review]
Looks good, just two minor comments. Can you also update the test reference result? Otherwise good to go.

**Review by MFraters — APPROVED** [review]

**Comment by MFraters** [comment]
hmm, I think you still need to include [include/world_builder/features/feature_utilities.h](https://github.com/GeodynamicWorldBuilder/WorldBuilder/blob/d6df754e5e5d0b14b14d080ff1ac06e84c1ceecb/include/world_builder/features/feature_utilities.h#L35)

**Comment by jdannberg** [comment]
I also looked at the test output of the changed test (in higher resolution):

comparison

The two main differences are that the very tip is now included as well in the sphere at the top (which is a rounding off question somewhere somehow I suspect), the other one is that the base of the plume tail is drawn to the max depth of the plume feature, even if it's below the deepest cross section given (which is a clear bugfix, because otherwise the max depth would be useless).

**Comment by MFraters** [comment]
Sounds good to me. I think it should be good to go as soon as you include [include/world_builder/features/feature_utilities.h](https://github.com/GeodynamicWorldBuilder/WorldBuilder/blob/d6df754e5e5d0b14b14d080ff1ac06e84c1ceecb/include/world_builder/features/feature_utilities.h#L35)

**Comment by jdannberg** [comment]
Okay, after several tries I think I now include all the things I need to include.

**Comment by MFraters** [comment]
Can you still add a changelog entry in a follow up pull request?

**Comment by jdannberg** [comment]
Sure. I thought this would just be a part of the plume feature. Do you want me to add it there or make a separate entry?

**Comment by MFraters** [comment]
hmm, it is adding a new temperature model to the plume feature, so in that sense I think it is a notable change. I will leave it up to you.

**Pull request merged**